docs/question/index.md

```diff
@@ -141,104 +178,54 @@ The `question.html` is a template used to render the question to the student. A
 
 The `question.html` is regular HTML, with four special features:
 
-1. Any text in double-curly-braces (like `{{params.m}}`) is substituted with variable values. If you use triple-braces (like `{{{params.html}}}`) then raw HTML is substituted (don't use this unless you know you need it). This is using [Mustache](https://mustache.github.io/mustache.5.html) templating.
+1. Any text in double-curly-braces (like `{{params.m}}`) is substituted with variable values. These values typically are defined by a [question's `server.py`](#custom-generation-and-grading-serverpy).
+2. Special HTML elements (like `<pl-number-input>`) enable input and formatted output. See the [list of PrairieLearn elements](../elements.md).
 
-2. Special HTML elements (like `<pl-number-input>`) enable input and formatted output. See the [list of PrairieLearn elements](../elements.md). Note that that **all submission elements must have unique `answers-name` attributes.** This is necessary for questions to be graded properly.
+   :warning: **All submission elements must have unique `answers-name` attributes.** This is necessary for questions to be graded properly.
 
 3. A special `<markdown>` tag allows you to write Markdown inline in questions.
-
 4. LaTeX equations are available within HTML by using `$x^2$` for inline equations, and `$$x^2$$` or `\[x^2\]` for display equations.
 
-## Question `server.py`
+!!! info
 
-## The `singleVariant` option for non-randomized questions
+    More information about `question.html` is in the [question.html documentation](html.md). Some non-intuitive aspects of the Markdown and LaTeX are described there.
 
-While it is recommended that all questions contain random parameters, sometimes it is impractical to do this. For questions that don't have a meaningful amount of randomization in them, the `info.json` file should set `"singleVariant": true`. This has the following effects:
+## Custom generation and grading (`server.py`)
 
-- On `Homework`-type assessments, each student will only ever be given one variant of the question, which they can repeatedly attempt without limit. The correct answer will never be shown to students.
-- On `Exam`-type assessments, all questions are effectively single-variant, so the `singleVariant` option has no effect.
+The `server.py` file for each question creates randomized question variants by generating random parameters and the corresponding correct answers. A minimal `server.py` might set `data["params"]` and `data["correct_answers"]`, and then use the built-in grading ability of the PrairieLearn elements to grade the students submission. A complete tutorial and example of a `server.py` file is described on the [server.py documentation](server.md) page.
```

Comment:
The last sentence here seems redundant with the below `!!! info` section.

Reply:
8de961fa8bacdf280e06dd8f5e73d124836d5b15